apps/circuit-compiler/src/app/app.tsx

```diff
@@ -1,17 +1,140 @@
-import React, { useEffect } from 'react'
+import React, {useEffect, useReducer, useState} from 'react'
+import {RenderIf} from '@remix-ui/helper'
+import {IntlProvider} from 'react-intl'
 
-import { CircomPluginClient } from './services/circomPluginClient'
+import { Container } from './components/container'
+import {CircuitAppContext} from './contexts'
+import {appInitialState, appReducer} from './reducers/state'
+import {CircomPluginClient} from './services/circomPluginClient'
+import { compileCircuit } from './actions'
 
 function App() {
+  const [appState, dispatch] = useReducer(appReducer, appInitialState)
+  const [plugin, setPlugin] = useState<CircomPluginClient>(null)
+  const [locale, setLocale] = useState<{code: string; messages: any}>({
+    code: 'en',
+    messages: null
+  })
+  const [isContentChanged, setIsContentChanged] = useState<boolean>(false)
 
   useEffect(() => {
-    new CircomPluginClient()
+    const plugin = new CircomPluginClient()
```

Comment:
setting plugin clients in useeffect and in a state will cause problems. especially when using strict mode. I've seen situations with other plugins doing this where it would either try to connect to Remix twice or never connect at all. You can can listen on the onload event here just fine. but I'd set the plugin as a const and export it.

Reply:
Can you make a commit with your recommended changes?

Reply:
sure